OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -831,42 +831,61 @@ private static DreamValue CreateValueFromJsonElement(JsonElement jsonElement)
             }
         }
 
+        /// <summary>
+        /// A helper function for /proc/json_encode(). Takes in a value and returns its json-able equivalent.
+        /// </summary>
+        /// <returns>Something that <see cref="JsonSerializer.Serialize"/> can parse.</returns>
         public static object CreateJsonElementFromValue(DreamValue value) {
-            if (value.TryGetValueAsFloat(out float floatValue)) {
+            return CreateJsonElementFromValueRecursive(value, true);
+        }
+
+        /// <remarks> This exists to allow for some control over the recursion.<br/>
+        /// <see langword="TODO:"/> Allow for deeper recursion while preventing infinite recursion and stack overflow</remarks>
+        private static object CreateJsonElementFromValueRecursive(DreamValue value, bool beRecursive) {
+            if(value.TryGetValueAsFloat(out float floatValue))
                 return floatValue;
-            } else if (value.TryGetValueAsString(out string text)) {
+            if (value.TryGetValueAsString(out string text))
                 return HttpUtility.JavaScriptStringEncode(text);
-            } else if (value.TryGetValueAsPath(out var path)) {
-                return path.ToString();
-            } else if (value.TryGetValueAsDreamList(out DreamList list)) {
-                if (list.IsAssociative()) {
+            if (value.TryGetValueAsPath(out var path))
+                return HttpUtility.JavaScriptStringEncode(path.ToString());
+            if (value.TryGetValueAsDreamList(out DreamList list)) {
+                List<Object> jsonArray = new();
+                if (!beRecursive) // TODO: handle recursion while preventing infinite recursion / stack overflow
+                    return jsonArray;
```

Comment:
So in it's current state, this won't encode lists inside lists? There are many places in SS13 code that require this to work. TGUI for example calls `json_encode()` on a list of data to be sent to clients, which itself contains lists.

Reply:
Maybe this is a bit *too* lazy, yeah. I can try to think up a quick functional fix.